### PR TITLE
attest to executed program

### PR DESCRIPTION
**Pull request by jan-ferdinand**
See [TIP 0006](https://github.com/TritonVM/triton-vm/blob/master/tips/tip-0006/tip-0006.pdf).

### CODE REVIEW

**Review by aszepieniec — APPROVED** [review]

File: specification/src/program-attestation.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+[blinding]: https://en.wikipedia.org/wiki/Blinding_(cryptography)
+[tip5]: https://eprint.iacr.org/2023/107.pdf
+
+[^zk]:
+Since hash functions are deterministic, a programmer desiring resistance against enumeration attacks might want to include [blinding elements][blinding]
+This is easily possible in, for example, dead code, by including instructions like `push $rand`.
+The distribution `$rand` is sampled from and the number of such instructions determine the security level against enumeration attacks.
+
```
[the remark this answers is not in the document]
Comment (jan-ferdinand) [reply specification/src/program-attestation.md:52]: Interesting, for me it renders without problem:

image

Which version of mdbook are on? Over here, it's v0.4.30.

[the remark this answers is not in the document]
Comment (jan-ferdinand) [reply specification/src/program-attestation.md:52]: In any case, my assumption is that the substring `$rand` appearing twice is the cause of the issue. The latex parser probably kicks in where it should not. I will get rid of that combination of characters.

[the remark this answers is not in the document]
Comment (aszepieniec) [reply specification/src/program-attestation.md:52]: Aha! I'm on mdbook v0.4.21.

File: specification/src/program-table.md [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
+
+Above procedure is iterated until the [necessary number of rows](arithmetization.md#padding) have been added.
+
+As an exception to all other [table-linking arguments](table-linking.md), the Program Table's instruction [Lookup Argument](lookup-argument.md) records the argument's initial value in the first row.
+This is necessary because an instruction's potential argument, or else the next instruction, is recorded in the next row.
+Hence, verifying correct initialization of the logarithmic derivative requires access to both the current and the next row.
+Only transition constraints can access two rows.
+Therefore, the initial recorded value of the logarithmic derivative must be independent of the second row.
+The logarithmic derivative's final value, allowing for a meaningful cross-table argument, is recorded in the first padding row.
+This row is guaranteed to exist due to the hash-input padding mechanics.
```
[the remark this answers is not in the document]
Comment (jan-ferdinand) [reply specification/src/program-table.md]: I rewrote the paragraph. Let me know what you think.

File: specification/src/hash-table.md [hunk 3]
```diff
@@ -74,205 +109,140 @@ Both types of challenges are X-field elements, _i.e._, elements of $\mathbb{F}_{
...
+1. For `i` $\in\{10, \dots, 15\}$:
+If the round number is 0 and the current `Mode` is `hash`, then register `state_i` is 1.
+1. For `i` $\in\{10, \dots, 15\}$:
+If the round number is 0 and the current instruction is `absorb_init`, then register `state_i` is 0.
+1. For `i` $\in\{0, \dots, 3\}$:
+let `state_i_hi_limbs_minus_2_pow_32` be 0 if and only if both of the two high limbs of `state_i` are their maximum value.
+That is, `state_i_hi_limbs_minus_2_pow_32` is an alias for the difference between $2^{32} - 1$ and the re-composed highest two limbs of `state_i`, _i.e._, $2^{16} \cdot{}$`state_i_highest_lk_in`${}+{}$`state_i_mid_high_lk_in`.
+    1. If the two high limbs of `state_i` are both the maximum possible value, then the two low limbs of `state_i` are both 0.
+    1. The `state_i_inv` is the inverse of `state_i_hi_limbs_minus_2_pow_32` or `state_i_inv` is 0.
+    1. The `state_i_inv` is the inverse of `state_i_hi_limbs_minus_2_pow_32` or `state_i_hi_limbs_minus_2_pow_32` is 0.
```
[the remark this answers is not in the document]
Comment (jan-ferdinand) [reply specification/src/hash-table.md:150]: I tried rephrasing and providing some more context. Let me know what you think.

**Review by aszepieniec — APPROVED** [review]

**Pull request merged**